pkg/apis/kubeflow.org/v2alpha1/trainjob_types.go

```diff
@@ -167,8 +167,9 @@ type DatasetConfig struct {
 	// These values will be merged with the TrainingRuntime's dataset initializer environments.
 	Env []corev1.EnvVar `json:"env,omitempty"`
 
-	// Reference to the TrainJob's secrets to download dataset.
-	SecretRef *corev1.SecretReference `json:"secretRef,omitempty"`
+	// Reference to the secret with credentials to download dataset.
+	// Secret must be created in the TrainJob's namespace.
+	SecretRef *corev1.LocalObjectReference `json:"secretRef,omitempty"`
```

Comment:
@tenzen-y I updated the API to use `*corev1.LocalObjectReference` type for the SecretRef.
I think, we should use this since we don't want to support cross-namespace reference for this secret.